### ISSUE TITLE
Kitsun integration broken on V6 of StreamDeck software

### ISSUE THREAD

**Issue by ascendedguard** [ascendedguard/streamdeck-japanesereviews#2]
Some change with the browser integration in V6 of the Stream Deck software now causes the two fetch requests made to Kitsun to not work as expected. The first login is successful, but the second call throws a 500 Internal Server Error on Kitsun's side.

**Comment by ascendedguard** [comment]
Just an update incase people come across this - I worked with Stream Deck developers a bit on this, and the response was basically "The Chrome engine changed this, we have no control over it so it's no longer possible."

I forget the details offhand since it's been a few months since I actually looked closely, but if I remember, Kitsun's login expects to set a cookie, and followup requests require the cookie. Login works, but fails to set the cookie because the Stream Deck executes from localhost via HTTP (not HTTPS), so the cookie is not set because the context is not secure. It also does not allow you to extract/set the cookies manually on requests.

So basically, to get Kitsun working again requires either a change from Kitsun to do proper API authentication instead of cookies, or the entire plugin must be written in another non-JS language. I was looking at doing this with C#, but the main challenges are getting the plugin working in the first place, but also redoing the dynamic graphics in C# in a way that makes the buttons look as good as the JS image manipulation, mostly the text rendering with shading behind.

**Comment by ascendedguard** [comment]
This is fixed in the latest builds, by converting the plugin to be Node.js based.
